File: Dante1/Module1.02/Zadanie2.03/main.c
#include <stdio.h>

int is_leap_year(unsigned year)
{
    return ((year & 3) == 0 && ((year % 25) != 0 || (year & 15) == 0));
}

int main ()
{
    int error;

    int day, month, year;
    printf("Enter: ");
    error = scanf("%2d-%2d-%4d" , &day, &month, &year);
    if (error != 3)
    {
        printf("Incorrect input");
        return 1;
    }

    printf("\n");
    if ((month == 4 && day == 31) || (month == 6 && day == 31) || (month == 9 && day == 31)
        || (month == 11 && day == 31) || (month == 2 && day > 28 && !is_leap_year(year)) ||
            1 > day || day > 31 || 1 > month || month > 12 || 999 > year)
    {
        printf("incorrect");
    }
    else
    {
        printf("correct");
    }

    return 0;
}

File: Dante1/Module1.02/Zadanie2.03/main.rs
use regex::Regex;
fn main() // i didnt do tests for checking month-day-year 'cause its too easy 
{
    println!("Enter: ");

    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Error in String input");

    let input = input.trim();

    let re = Regex::new(r"^(\d+)-(\d+)-(\d+)$").expect("Error in regex transformation");

    if let Some(caps) = re.captures(input)
    {
        let day: i32 = caps[1].parse().expect("Day number");
        let month: i32 = caps[2].parse().expect("Month number");
        let year: i32 = caps[3].parse().expect("Year number");

        println!("Day number: {}", day);
        println!("Month number: {}", month);
        println!("Year number: {}", year);
    }
    else
    {
        println!("Incorrect input");
    }
}
